### PR TITLE
Fold Until implementation, may need more work to get right.

### PR DESCRIPTION
**Pull request by rtkelly13**
But is a much needed addition. Composes very well using Option.FoldUntil
Need to see if some of the fold methods are actually Reduce methods (Wrong name)
Some mutable iteration code in there, but everything else is pretty clean so far.

### CODE REVIEW

**Comment by rtkelly13** [comment]
Implementation of https://github.com/Resultful/Resultful/issues/2

@chartjunk could you have a look at this.
I think it's a bit rough so far.

File: Resultful/ResultExtensions.cs [hunk 1]
```diff
@@ -143,6 +170,21 @@ public static Result<T> Fold<T>(this IEnumerable<Result<T>> values, Func<T, T, T
             TResult seed, Func<TResult, T, TResult> aggrFunc)
             => values.Fold(seed.Ok(), aggrFunc);
 
+        //FoldUntil on Result<T>
+        public static Result<T> FoldUtil<T>(this IEnumerable<Result<T>> values, Func<T, T, T> plusFunc)
```
Comment (chartjunk) [inline Resultful/ResultExtensions.cs]:
```suggestion
        public static Result<T> FoldUntil<T>(this IEnumerable<Result<T>> values, Func<T, T, T> plusFunc)
```

File: Resultful/ResultExtensions.cs [hunk 2]
```diff
@@ -143,6 +170,21 @@ public static Result<T> Fold<T>(this IEnumerable<Result<T>> values, Func<T, T, T
...
+            => values.ThrowIfDefault(nameof(values)).Aggregate((acc, value) =>
+                acc.Bind(x => value.Map(y => plusFunc.ThrowIfDefault(nameof(plusFunc))(x, y))));
+
+        public static Result<TResult> FoldUtil<TResult, T>(this IEnumerable<Result<T>> values,
```
Comment (chartjunk) [inline Resultful/ResultExtensions.cs]:
```suggestion
        public static Result<TResult> FoldUntil<TResult, T>(this IEnumerable<Result<T>> values,
```

File: Resultful/ResultExtensions.cs [hunk 3]
```diff
@@ -143,6 +170,21 @@ public static Result<T> Fold<T>(this IEnumerable<Result<T>> values, Func<T, T, T
...
+            Result<TResult> seed, Func<TResult, T, TResult> aggrFunc)
+            => values.ThrowIfDefault(nameof(values)).Aggregate(seed, (acc, value) =>
+                acc.Bind(x => value.Map(y => aggrFunc.ThrowIfDefault(nameof(aggrFunc))(x, y))));
+
+        public static Result<TResult> FoldUtil<TResult, T>(this IEnumerable<Result<T>> values,
```
Comment (chartjunk) [inline Resultful/ResultExtensions.cs]:
```suggestion
        public static Result<TResult> FoldUntil<TResult, T>(this IEnumerable<Result<T>> values,
```

File: Resultful/ResultExtensions.cs [hunk 4]
```diff
@@ -143,6 +170,21 @@ public static Result<T> Fold<T>(this IEnumerable<Result<T>> values, Func<T, T, T
...
+            TResult seed, Func<TResult, T, TResult> aggrFunc)
+            => values.FoldUtil(seed.Ok(), aggrFunc);
```
Comment (chartjunk) [inline Resultful/ResultExtensions.cs]:
```suggestion
            => values.FoldUntil(seed.Ok(), aggrFunc);
```

**Comment by rtkelly13** [comment]
The whole library could do with more tests 😅
My lack of even basic spelling is also on full display here, copy/paste making it worse

**Pull request merged**